pyteal/ast/router_test.py

```diff
@@ -528,8 +530,29 @@ def test_contract_json_obj():
     router = pt.Router(contract_name, on_complete_actions)
     method_list: list[sdk_abi.Method] = []
     for subroutine in abi_subroutines:
-        router.add_method_handler(subroutine)
-        method_list.append(sdk_abi.Method.from_signature(subroutine.method_signature()))
+
+        doc = subroutine.subroutine.implementation.__doc__
+        desc = None
+        if doc is not None and doc.strip() == "replace me":
+            desc = "dope description"
+
+        router.add_method_handler(subroutine, description=desc)
+
+        ms = subroutine.method_spec()
+
+        # Manually replace it since the override is applied in the method handler
+        # not attached to the ABIReturnSubroutine itself
+        ms.desc = desc if desc is not None else ms.desc
+
+        sig_method = sdk_abi.Method.from_signature(subroutine.method_signature())
+
+        assert ms.name == sig_method.name
```

Comment:
shouldn't there be an assertion involving  `sig_method.desc` ?

Reply:
sig_method doesnt get a description since it just parses out the `methodname(arg,arg)`